\chapter{Introduction}
\label{cha:intro}

It is astonishing how, based on the same genetic information coded into the DNA, cells in an organism differentiate into a variety of specialized types with a multitude of different tasks. 
Beginning with a single cell, a temporal, spatial and functional coordination is crucial during the body formation of virtually any living organism on this planet. 
Acting as an additional regulatory layer, the epigenome describes a set of chemical modifications to the DNA, controlling the activation and transcription of genes into RNA and ultimately proteins.




\section{Motivation}
\label{sec:intro:motivation}

Starting 1990 and taking over a decade until completion, the human genome project incorporated an international team of researchers with the aim of deciphering the genetic blueprint to build a human being. Based on elementary sequencing technology, the outcome was already a nearly complete reference sequence including gene annotations.
Since then, the development of high-throughput sequencing technologies enabled studies of countless organisms, cell types and disease conditions.

Most recently, a third-generation of sequencing techniques are introducing new perspectives to the field of genome analysis by generating previously unattainable read lengths with averages in the tens of thousands of nucleotides.
Portable nanopore sequencing devices producing large data sets within few days, open a new field of research at the intersection of genomics, computer science and engineering. New data types, formats and error characteristics demand here the adaptation of existing and development of new algorithms for bioinformatic software.




\section{Biological Background}
\label{sec:intro:bio}

\begin{figure}[h]
	\centering
	\includegraphics[width=1.0\textwidth]{figures/intro/chromatin.pdf}
	\captionsetup{format=plain}
	\caption[Structure]{Chromosome to nucleotide structure (Adapted from \cite{zymo2020})}
	\label{fig:intro:chromatin}
\end{figure}

\begin{figure}[h]
	\centering
	\includegraphics[width=1.0\textwidth]{figures/intro/methylation.pdf}
	\captionsetup{format=plain}
	\caption[DNA methylation dynamics]{Methylation dynamics during development (Adapted from \cite{Greenberg2019})}
	\label{fig:intro:methylation}
\end{figure}

\begin{itemize}
    \item Organism, cell, DNA
    \item Epigenetics as additional layer
    \begin{itemize}
        \item Briefly histone modifications
        \item DNA methylation, 5mC, hmC, 6mA
    \end{itemize}
    \item Without explicit statement: Focus of the Meissner lab
\end{itemize}




\section{Technical Background}
\label{sec:intro:sequencing}

\begin{figure}[h]
	\centering
	\includegraphics[width=1.0\textwidth]{figures/intro/sbs.pdf}
	\captionsetup{format=plain}
	\caption[Sequencing by synthesis]{Sequencing by synthesis}
	\label{fig:intro:sbs}
\end{figure}

\begin{figure}[h]
	\centering
	\includegraphics[width=1.0\textwidth]{figures/intro/long_read.pdf}
	\captionsetup{format=plain}
	\caption[Long read sequencing]{PacBio SMRT and ONT nanopore sequencing.}
	\label{fig:intro:longread}
\end{figure}

\begin{itemize}
    \item Sequencing
    \item Helene: "Easy to forget: but how to calculate a methylation rate from multiple reads and why minimum coverage. There is even a ref for 10X."
    \item Breifly generations, 1st, 2nd, 3rd ...
    \item Detail 3rd generation
    \begin{itemize}
        \item Briefly PacBio, Roche
        \item Detail ONT
    \end{itemize}
\end{itemize}

Among other devices distributed by Oxford Nanopore Technologies (ONT), the MinION in particular is gaining prominence. In brief, the nanopore sequencing process is based on guiding a nucleotide polymer through a pore inserted in a membrane while measuring a change in ionic current as a proxy signal over time. This signal is then interpreted to determine the underlying DNA or RNA sequence. The nanopore technology enables direct readout of sequences from individual DNA or RNA molecules, including base modifications, since no synthesis or amplification is required.




\section{Structure}
\label{sec:intro:structure}

Briefly describe the overall structure of the thesis, from literature to pipeline, signal processing and finally repeat detection.

%\textbf{Chapter \ref{sec:nanopype}} \\[0.2em]
%\blindtext

