\chapter{Introduction}
\label{cha:intro}

It is astonishing how, based on the same genetic information coded into the DNA, cells in an organism can differentiate into a variety of specialized types with a multitude of different tasks.
Beginning with a single cell, a temporal, spatial and functional coordination determines the growth and body formation of eukaryotic organisms.
Depending on the cell state, only a fraction of genes is actively transcribed, while others remain repressed.
Acting as an additional regulatory layer, the epigenome describes a set of chemical modifications made to the DNA, controlling, inter alia, the activation and transcription of genes into RNA and ultimately proteins.
Present sequencing technologies and connected bioinformatics provide researchers with the tools to study sequence and epigenetic state down to single cell and single molecule resolution.
Starting 1990 and taking over a decade until completion, the human genome project incorporated an international team of researchers with the aim of deciphering for the first time the genetic blueprint to build a human being. 
Based on elementary sequencing technology, the outcome was already a nearly complete reference sequence including gene annotations.
Since then, the development of high-throughput, next-generation sequencing (NGS) technologies enabled studies of countless organisms, cell types and disease conditions. 
While being very reliable in terms of throughput and accuracy, sequenced fragments of at most few hundreds nucleotides in length still limit the readout from repetitive regions or resolution of long distance dependencies on a single molecule.




\section{Motivation}
\label{sec:intro:motivation}

Most recently, a third-generation of sequencing techniques is introducing new perspectives to the field of genome analysis by generating previously unattainable read lengths with averages in the tens of thousands of nucleotides. 
Still under active development and with frequent improvements, long-read sequencing provides new opportunities by, visually speaking, increasing the size of the puzzle pieces.
Moreover, direct sequencing of DNA and RNA molecules using the nanopore technology facilitates the detection of different types of base modifications.
Modern nanopore sequencing devices, producing large data sets within few days, open therefore a new field of research at the intersection of genomics, computer science and engineering. 
New data types, formats and error characteristics demand here the adaptation of existing and development of new algorithms for bioinformatic software.




\section{Genome Regulation}
\label{sec:intro:bio}

Most cell types of an eukaryotic organism contain a copy of the genetic code in form of folded DNA in the nucleus.
Virtually all mammals have diploid cells with homologous maternal and paternal copies, organized into chromosome pairs with the same genes at the same locations.
While the entire human genome comprises about three billion nucleotides in total, the longest continuous stretch of DNA is the first of 23 chromosome pairs with 247 million base pairs. 
To maintain its integrity and supporting the chromatin structure formation, the DNA is wrapped around octamers of histone proteins called nucleosomes (Fig. \ref{fig:intro:chromatin}).


\begin{figure}[h]
	\centering
	\includegraphics[width=1.0\textwidth]{figures/intro/chromatin.pdf}
	\captionsetup{format=plain}
	\caption[Chromosome to nucleotide structure]{The DNA of each chromosome is wrapped around histones, forming nucleosomes and the chromatin structure. Chemical modifications to histone tails and individual base pairs impact properties and cellular function (Adapted from \cite{zymo2020}).}
	\label{fig:intro:chromatin}
\end{figure}

The \textasciitilde 147bp of DNA directly twisted around each nucleosome is protected against physical access from for instance transcription factors (TFs). 
The dynamic positioning of nucleosomes is therefore part of the epigenetic regulation process during transcription and replication.
Accordingly, nucleosomes are enriched at compacted heterochromatin in mostly inactive genomic regions, while being depleted at regulatory elements such as enhancers, insulators and transcribed genes \cite{Klemm2019}. 
A growing number of chemical modifications to histone tails (H2A, H2B, H3 and H4) is reported to impact inter-nucleosomal interactions, but also leading to recruitment of proteins and complexes involved in transcription, replication or DNA repair \cite{Bannister2011}. 
An example is the methylation of the lysine 79 of the H3 tail (H3K79me3) found at the transcription start sites (TSS) of active genes \cite{Lawrence2016}. 
To be further mentioned in this context are H3K4me3 marking active euchromatin, while H3K9me3 is indicating repressive heterochromatin.
Gene expression is further influenced by active enhancer sites with enrichment of H3K27ac (acetylation).
Finally, repressed and bivalent promoters are modified with H3K27me3 and the combination of H3K4me4 and H3K27me3 respectively.



Roles of DNA methylation \cite{Greenberg2019}

\begin{figure}[h]
	\centering
	\includegraphics[width=1.0\textwidth]{figures/intro/cancer.pdf}
	\captionsetup{format=plain}
	\caption[DNA methylation in cancer]{}
	\label{fig:intro:cancer}
\end{figure}

repression of transposable elements LINE1 and IAP
hemi-methylation on genomic imprinting, ICRs
X-chromosome inactivation
high in actively transcribed gene bodies

mutagenic, 5mC to T transition, reduced CG content in these organisms
29.4M (instead of 188M) in hg38
21.8M (instead of 169M) in mm10

DNMT3A/B de-novo methylation
DNMT1 maintenance

passive de-methylation during replication
active de-methylation by TET1-3

loss of methylation is lethal in early embryonic development
de-regulation in cancer


\begin{figure}[h]
	\centering
	\includegraphics[width=1.0\textwidth]{figures/intro/methylation.pdf}
	\captionsetup{format=plain}
	\caption[DNA methylation dynamics]{Methylation dynamics during development (Adapted from \cite{Greenberg2019})}
	\label{fig:intro:methylation}
\end{figure}

The road ahead \cite{McGuire2020}




\section{Sequencing Technologies}
\label{sec:intro:sequencing}

\begin{figure}[h]
	\centering
	\includegraphics[width=1.0\textwidth]{figures/intro/sbs.pdf}
	\captionsetup{format=plain}
	\caption[Sequencing by synthesis]{Sequencing by synthesis}
	\label{fig:intro:sbs}
\end{figure}

Third revolution review \cite{Dijk2018}

\begin{figure}[h]
	\centering
	\includegraphics[width=1.0\textwidth]{figures/intro/long_read.pdf}
	\captionsetup{format=plain}
	\caption[Long read sequencing]{PacBio SMRT and ONT nanopore sequencing.}
	\label{fig:intro:longread}
\end{figure}

\begin{itemize}
    \item Sequencing
    \item Helene: "Easy to forget: but how to calculate a methylation rate from multiple reads and why minimum coverage. There is even a ref for 10X."
    \item Breifly generations, 1st, 2nd, 3rd ...
    \item Detail 3rd generation
    \begin{itemize}
        \item Briefly PacBio, Roche
        \item Detail ONT
    \end{itemize}
\end{itemize}

Among other devices distributed by Oxford Nanopore Technologies (ONT), the MinION in particular is gaining prominence. In brief, the nanopore sequencing process is based on guiding a nucleotide polymer through a pore inserted in a membrane while measuring a change in ionic current as a proxy signal over time. This signal is then interpreted to determine the underlying DNA or RNA sequence. The nanopore technology enables direct readout of sequences from individual DNA or RNA molecules, including base modifications, since no synthesis or amplification is required.




\section{Structure}
\label{sec:intro:structure}

Briefly describe the overall structure of the thesis, from literature to pipeline, signal processing and finally repeat detection.

%\textbf{Chapter \ref{sec:nanopype}} \\[0.2em]
%\blindtext

