\chapter{State of the Art}
\label{sec:state_of_art}

\begin{itemize}
    \item Zoom out history \cite{Deamer2016}
\end{itemize}

\section{Nanopore Sequencing in the Literature}

\begin{itemize}
    \item Motivation: Too many publications to read, too fast progress to stay up to date
    \item Motivation: OpenSyllabusGallxy
    \item Introduce S2 dataset, summary, paper, journals, topics?
    \item Alternative: REST API?
    \item Plot UMAP of all publications, highlight 'nanopore cluster'
    \item paper over time, impact (over time?) topics
    \item wordclouds
    \item most impact by incoming citations, normalized by year/age, cluster-center?
    \item missing but relevant publications for this work, limitations of computational literature scan
\end{itemize}

\section{\Biorxiv\ Preprints}

\begin{itemize}
    \item Role of preprints for nanopore, bioRxiv as platform
    \item stats of bioRxiv, paper, topics?, nanopore paper
    \item projection of bioRxiv to S2, overlay with previous cluster
    \item Example of read-until, visible for in-field experts, hidden for outsider?
    \item Comment/Opinion on preprints in fast-moving fields, coronavirus example, weight potential against risks
    \item Mention Twitter as fast but also least reliable source
\end{itemize}

\section{Throughput and Accuracy}

\begin{itemize}
    \item Representative Flow-Cells over time
    \item Size selection, shearing, nuclease flush
    \item throughput over time plot
    \item accuracy last albacore, guppy-fast, guppy-hac (mention flappie, scrappie, SACall, chiron, ...)
    \item cite benchmark paper
    \item Mapping of long reads depending on genomic context, read length, Q-Score etc.
    \item Accuracy of base modifications: Nanopolish, Signalign, DeepMod, DeepSignal etc.
    \item Nanopolish context and signal level analysis
\end{itemize}