\chapter{Discussion}
\label{cha:summary}

\textit{But could you not also do that with short reads?} - Expressing an initial scepticism against a novel technology, nanopore sequencing is, six years after the first release in 2014, still termed the emerging technology.
Next-generation sequencing remains the established and widely undisputed reference method.
However, after significant improvements, addressing shortcomings of throughput and accuracy, both third-generation platforms have evolved to become the primary technology for genome assembly and are commonly used for the analysis of viral and bacterial genomes.
Yet, a broader replacement of NGS methods is not in sight, raising the question, why the introductory question is not more frequently re-phrased to: \textit{Would nanopore here not be the more suitable platform?} Or: \textit{Would you not confirm this by using nanopore sequencing?}

The underlying research question of this work is therefore to analyze unique use-cases for nanopore sequencing, identify key limitations and provide a comprehensive investigation of the current status.
Despite many improvements, the application of nanopore sequencing in the context of large mammalian and plant genomes remains challenging.
At the same time are the benefits of long reads known: Reliable alignments in repetitive regions, no duplicates and biases by library amplification and the linkage of distant genetic and epigenetic features on single molecule level.
Yet, the stable throughput, higher read accuracy and established workflows of NGS technologies justify the commitment to nanopore sequencing only for applications infeasible with short reads.

A niche to gain traction at applying nanopore sequencing is the analysis of repetitive regions in the human genome.
For comparison, using short reads, the reachable part of a repeat is limited by the read length, leading to ambiguous alignments, once a read contains only repeat sequence.
Short tandem repeats (STRs) are accumulations of three to six nucleotide long sequence patterns, in disease cases expanded to hundreds of copies.
The epigenetic analysis of repeats is a unique feature of nanopore sequencing, considering that SMRT sequencing is not sufficiently sensitive to 5-methylcytosine on single-molecule level.
Repeat detection by sequencing is the digital advancement over southern blot based quantification, increasing the accuracy and reducing the turnaround time.
\textit{STRique} is a bioinformatic analysis software, developed to integrate the quantification of short tandem repeats with their methylation state on individual read level.
The algorithmic key features of \textit{STRique} are raw signal alignment and the sequence to signal annotation.
Due to the oversampling of the sequencer, raw signal traces are roughly ten times longer compared to their corresponding sequence and, with respect to noise and time warping, generally more difficult to handle.
Yet, only a signal based counting algorithm allows to bypass systematic errors induced during conventional basecalling of tandem repeats.
Hidden Markov Models (HMM) are a powerful resource to align nanopore signals to a reference sequence, with the most prominent usage in the \textit{Nanopolish} package.
However, their computational complexity is scaling quadratically with the number of hidden states, limiting the usage to restricted signal and sequence windows of interest.
The purpose of the signal alignment is therefore, to extract a signal segment covering the repeat with sufficient flanking sequence to anchor a counting HMM.
The flexibility of the HMM is needed over a static alignment, since the repeat length is unknown and varying between reads.
A looping transition around a single repeat profile allows the model to iterate through an arbitrary number of repeat patterns.
The signal alignment is the enabling and at the same time limiting step: The Viterbi path through the HMM is the most likely state-sequence given the observed signal, fed with a wrong signal alignment, the model will still yield a repeat count.
A filtering based on signal alignment scores is therefore crucial to exclude low quality counts.
A useful side-effect of the signal alignment is the ability to mask regions within the read.
\textit{STRique} provides a script to cut out the expanded repeat from the raw signal and creates wild-type like reads passing any conventional downstream pipeline, to detect for instance DNA-methylation in the repeat-flanking sequence.
Taken together, \textit{STRique} is an example for a unique nanopore application, enabling more detailed investigations of short tandem repeat expansions.

Bioinformatics for nanopore sequencing are undergoing constant development.
Even a stable nanopore workflow needs, in contrast to NGS analysis, constant maintenance to factor in accuracy improvements, new pore generations, but also for instance recent file format and compression method changes.
Zooming out from individual tool to pipeline level, the streamlined processing and data archiving has received very little attention in the nanopore community.
Computational setups ranging from few MinION flow cells analyzed on a Laptop to high-throughput PromethION sequencing in combination with GPU accelerated cluster environments are complicating, if not inhibiting, a uniform processing strategy.
The motivation behind the development of the \textit{Nanopype} pipeline is therefore not only to wrap the processing, but also set up a scaling data management.
While certainly site specific on the implementation detail, the concepts of the pipeline backend transfer to any institute aiming to set up a nanopore workflow.
The ONT device control software \textit{MinKnow} is providing raw data in batches of currently 4k reads packed into \textit{fast5} files.
Further transfer to permanent storage, indexing and archiving are essential for a high-throughput environment.
The organization in batches is, as opposed to a single \textit{fastq} file per NGS run, beneficial for distributed processing.
\textit{Nanopype} operates on batches whenever possible, submitting for instance alignments in batches bears the advantage that multiple compute nodes can process a single sequencing run and that only few reads need to be re-processed in case of errors.

Using pipeline management systems such as \textit{Nextflow} or \textit{Snakemake} over plain shell script implementations


Nanopype: Processing to stay up to date with updating software
storage to handle raw data, x times larger than fastq from NGS
snakemake over shell scripts for modularity, supporting container, multiple cluster backends
Build and functionality tests are very rare!
Interchangeable tools for lab focused on development
weakness: trying to maintain increasing number of integrated tools and their dependencies, example pytorch/tensorflow v1/v2 for neural networks
Future work: freeze core pipeline, develop NanopypeXT which uses Nanopype as sub-workflow for very specific or complex to maintain tools

Literature: Throughput and accuracy increasingly suitable for application to large plant and mammalian genomes
Assembly and structural variant detection are established but will further benefit from accuracy and even longer reads
Limitations on wet-lab side with high molecular weight DNA and amount of input material
Limitations on bioinformatic, storage and upload, refer to STRique data being in Figshare repository
Base modification visibility long known, lack of robust and accurate tools
RNA modifications example of frequently mentioned potentials with only few applications up to date

Future work based on pipeline backend and baseline set with signal annotation
Base modification framework on event data without dependency to basecaller generation
Base caller based on novel transformer neural networks with impressive results in translation and speech to text







